### DISCUSSION TITLE
Use console.debug instead of wrapping console.log?

### DISCUSSION THREAD

**Discussion by baodrate** [scambier/obsidian-omnisearch#417] [Ideas]
omnisearch implements debug logging by having a plugin setting that toggles the behavior of `logDebug`, which wraps `console.log`.

https://github.com/scambier/obsidian-omnisearch/blob/1c98d8b2d7f84eaffb0a1f1871f87d5532e84713/src/tools/utils.ts#L233-L252

Why not just call `console.debug` instead? Then you don't need a separate setting, and you can filter debug level log messages dynamically (they don't get dropped, they're just hidden by default)

<img width="621" alt="image" src="https://github.com/user-attachments/assets/3cffdea9-e002-468c-9ff0-bb382ff83bed">

And by calling `console.debug`, the developer console will show the line where the log message was called, instead of the definition of `logDebug`

Also, wrapping `console.log` to prepend a timestamp is unnecessary because you just need to enable console timestamps in the developer console:

<img width="522" alt="image" src="https://github.com/user-attachments/assets/54be0b77-77d8-4d8b-92a4-e496774a44ed">

<img width="835" alt="image" src="https://github.com/user-attachments/assets/6c53bea5-bef4-40e4-ac59-50657e902ec6">

**Comment by baodrate** [comment]
On top of that. I think that Omnisearch logs too much by default. From the [plugin guidelines](https://docs.obsidian.md/Plugins/Releasing/Plugin+guidelines#Avoid+unnecessary+logging+to+console):

> Please avoid unnecessary logging.
> In it's default configuration, the developer console should only show error messages, debug messages should not be shown.

**Comment by scambier** [comment]
Good points.

Edit: actually, `logDebug()` and `warnDebug()` are useful because they check if the verbose logging is enabled. Though I'll replace the `console.log` call with a `console.debug`.
